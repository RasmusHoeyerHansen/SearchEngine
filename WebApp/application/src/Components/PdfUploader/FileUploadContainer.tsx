import {ChangeEvent, useState} from "react";
import {FileUploadForm} from "./FileUploadForm";



const UploadedFile = () : JSX.Element => {
    return <div role = "uploaded files"/>;
}

export const FileUploadContainer = () => {
    const [pdfFiles, setState] = useState<File[]>()

    let displayText = "Upload a PDF file to contribute to the search engine."
    const onFileSelectionChanged = (event: ChangeEvent<HTMLInputElement>) => {
        let selectedFiles: File[] = [];
        const element = event.currentTarget as HTMLInputElement;

        let fileList: FileList | null = element.files;
        if (fileList) {
            for (let itm in fileList) {
                let item: File = fileList[itm];
                if (!selectedFiles.includes(item))
                    selectedFiles.push(item);
            }
        }
        setState(selectedFiles);
    }

    const onFileSubmit  = (event : SubmitEvent) => {


    }

    return (<div className={"PdfUploadContainer"} data-testid='FileUploadContainer'>
        <FileUploadForm onSubmit={onFileSubmit} onChange={onFileSelectionChanged}/>
        <UploadedFile />
    </div>)
}

export default FileUploadContainer;