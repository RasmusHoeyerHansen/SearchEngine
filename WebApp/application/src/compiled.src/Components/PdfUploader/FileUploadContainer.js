var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import { jsx as _jsx, jsxs as _jsxs } from "react/jsx-runtime";
import { useState } from "react";
import { FileUploadForm } from "./FileUploadForm";
var UploadedFile = function () {
    return _jsx("div", { role: "uploaded files" }, void 0);
};
export var FileUploadContainer = function () {
    var _a = useState(), pdfFiles = _a[0], setState = _a[1];
    var displayText = "Upload a PDF file to contribute to the search engine.";
    var onFileSelectionChanged = function (event) {
        var selectedFiles = [];
        var element = event.currentTarget;
        var fileList = element.files;
        if (fileList) {
            for (var itm in fileList) {
                var item = fileList[itm];
                if (!selectedFiles.includes(item))
                    selectedFiles.push(item);
            }
        }
        setState(selectedFiles);
    };
    var onFileSubmit = function (event) {
    };
    return (_jsxs("div", __assign({ className: "PdfUploadContainer", "data-testid": 'FileUploadContainer' }, { children: [_jsx(FileUploadForm, { onSubmit: onFileSubmit, onChange: onFileSelectionChanged }, void 0), _jsx(UploadedFile, {}, void 0)] }), void 0));
};
export default FileUploadContainer;
