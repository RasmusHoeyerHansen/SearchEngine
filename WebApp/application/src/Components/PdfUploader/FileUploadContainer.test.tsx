import React, {FormEvent} from "react";
import {
    cleanup,
    fireEvent,
    render,
    screen
} from "@testing-library/react";
import FileUploadContainer from "./FileUploadContainer";


describe("FileUploadContainer", () => {


    beforeEach(() => {
        jest.clearAllMocks();
        cleanup();
    });
    it('Matches snapshot', function () {
        let component = CreateComponent();
        expect(component).toMatchSnapshot();
    });

    it('should call submit on submit click', function () {
        const file = new File([new ArrayBuffer(1)], 'testfile.pdf');
        let mockFunction = jest.fn()
        let content = CreateComponent();
        fireEvent.change(content, {
            target: {files: [file]}
        });
    });

    interface testProps {
        onSubmit?: (event:SubmitEvent) => void;
        onChange?:(event:FormEvent<HTMLInputElement> ) => void
    }

    const CreateComponent = () => {
        render(<FileUploadContainer />);
        return screen.getByTestId('FileUploadContainer') as HTMLInputElement;
    };
})