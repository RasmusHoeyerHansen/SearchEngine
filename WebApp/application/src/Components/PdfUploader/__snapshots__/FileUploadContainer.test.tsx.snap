// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileUploadContainer Matches snapshot 1`] = `
<div
  class="PdfUploadContainer"
  data-testid="FileUploadContainer"
>
  <form
    data-testid="FileUploadForm"
    id="FileUploadForm"
    role="form"
  >
    <input
      accept="application/pdf"
      data-testid="FileInput"
      id="FileUploadForm-input"
      multiple=""
      name="formFile"
      role="input"
      type="file"
    />
  </form>
  <div
    role="uploaded files"
  />
</div>
`;
